src/main/scala/org/apache/spark/sql/hive/llap/LlapExternalCatalog.scala

```diff
@@ -123,47 +122,21 @@ private[spark] class LlapExternalCatalog(
       table: String,
       ignoreIfNotExists: Boolean,
       purge: Boolean): Unit = withClient {
-    requireDbExists(db)
-    val sessionState = SparkSession.getActiveSession.get.sessionState.asInstanceOf[LlapSessionState]
-    val stmt = sessionState.connection.createStatement()
-    val ifExistsString = if (ignoreIfNotExists) "IF EXISTS" else ""
-    val purgeString = if (purge) "PURGE" else ""
-    stmt.executeUpdate(s"DROP TABLE $ifExistsString $db.$table $purgeString")
+    if (Thread.currentThread().getStackTrace()(5).toString().contains("CreateHiveTableAsSelectCommand")) {
```

Comment:
Also, this fixes the case `CreateHiveTableAsSelectCommand` fails to clean up the empty table on failure.